File: podapi_ts/src/Models/LookupItem.ts
import { JsonProperty } from "json-object-mapper";
import LookupItemDTO from "./DTOs/LookupItemDTO";
import Model from "./Model";

// tslint:disable-next-line:interface-name
export interface ILookupItem {
  trackId: number;
  artistName: string;
  trackName: string;
  feedUrl: string;
  artworkUrl100: string;
  artworkUrl600: string;
  releaseDate: string;
  genres: string[];
  genreIds: string[];
}

export class LookupItem extends Model<LookupItemDTO> implements ILookupItem {
  @JsonProperty()
  public trackId: number;

  @JsonProperty()
  public artistName: string;

  @JsonProperty()
  public trackName: string;

  @JsonProperty()
  public feedUrl: string;

  @JsonProperty()
  public artworkUrl100: string;

  @JsonProperty()
  public artworkUrl600: string;

  @JsonProperty()
  public releaseDate: string;

  @JsonProperty()
  public genres: string[];

  @JsonProperty()
  public genreIds: string[];

  constructor(dto: LookupItemDTO) {
    super(dto);
  }

  public mapFromDTO(dto: LookupItemDTO): void {
    this.trackId = dto.trackId;
    this.artistName = dto.artistName;
    this.trackName = dto.trackName;
    this.feedUrl = dto.feedUrl;
    this.artworkUrl100 = dto.artworkUrl100;
    this.artworkUrl600 = dto.artworkUrl600;
    this.releaseDate = dto.releaseDate;
    this.genres = dto.genres;
    this.genreIds = dto.genreIds;
  }
}

File: podcast_app/lib/core/models/lookup_item.dart
import 'package:equatable/equatable.dart';
import 'package:flutter/foundation.dart';

class LookupItem extends Equatable {
  final String trackId;
  final String artistName;
  final String trackName;
  final String feedUrl;
  final String artwork100;
  final String artwork600;
  final String releaseDate;
  final List<String> genres;
  final List<String> genreIds;

  LookupItem({
    @required this.trackId,
    @required this.artistName,
    @required this.trackName,
    @required this.feedUrl,
    @required this.artwork100,
    @required this.artwork600,
    @required this.releaseDate,
    @required this.genres,
    @required this.genreIds,
  });

  static LookupItem fromJson(Map<String, dynamic> map) {
    return LookupItem(
      trackId: map['trackId'],
      artistName: map['artistName'],
      trackName: map['trackName'],
      feedUrl: map['feedUrl'],
      artwork100: map['artwork100'],
      artwork600: map['artwork600'],
      releaseDate: map['releaseDate'],
      genres: map['genres'].map<String>((i) => i.toString()).toList(),
      genreIds: map['genreIds'].map<String>((i) => i.toString()).toList(),
    );
  }

  @override
  List<Object> get props => [trackId];
}
